pkg/cmd/update-tls-artifacts/metadata/violation.go

```diff
@@ -0,0 +1,72 @@
+package metadata
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+)
+
+const unknownOwner = "Unknown"
+
+type Violation struct {
+	Name     string
+	Markdown []byte
+	Registry *certgraphapi.PKIRegistryInfo
+}
+
+func (v Violation) getJSONFilePath(parentDir string) string {
+	return filepath.Join(parentDir, fmt.Sprintf("%s-violations.json", v.Name))
+}
+
+func (v Violation) getMarkdownFilePath(parentDir string) string {
+	return filepath.Join(parentDir, fmt.Sprintf("%s.md", v.Name))
+}
+
+func (v Violation) DiffWithExistingJSON(parentDir string) error {
```

Comment:
If you're wrapping in this way, is it more logical to simply have a `DiffWithExisting` that does all the pertinent files?